tests/cpptests/test_cpp_query.cpp

```diff
@@ -447,15 +447,17 @@ TEST_F(QueryTest, testParser_v3) {
   // Tag queries
   assertValidQuery("@tags:{foo}", ctx);
   assertValidQuery("@tags:{foo|bar baz|boo}", ctx);
-  assertValidQuery("@tags:{foo|bar\\ baz|boo}", ctx);
+  assertInvalidQuery("@tags:{foo|bar\\ baz|boo}", ctx);
   assertValidQuery("@tags:{foo*}", ctx);
   assertValidQuery("@tags:{foo\\-*}", ctx);
   assertValidQuery("@tags:{bar | foo*}", ctx);
-  assertValidQuery("@tags:{bar* | foo}", ctx);
-  assertValidQuery("@tags:{bar* | foo*}", ctx);
+  assertInvalidQuery("@tags:{bar* | foo}", ctx);
```

Comment:
Please test that it is valid when star is escaped

Reply:
OK. Done. Thanks.